File: php/spec/comma_separated.spec.php
<?php declare(strict_types=1);

use Enotype\Loaders;

describe('commaSeparated', function() {
  describe('with \'one,two,three\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::commaSeparated('one,two,three'))->toEqual(['one', 'two', 'three']);
    });
  });
  
  describe('with \' one,two,three \'', function() {
    it('produces the expected result', function() {
      expect(Loaders::commaSeparated(' one,two,three '))->toEqual(['one', 'two', 'three']);
    });
  });
  
  describe('with \'one , two , three\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::commaSeparated('one , two , three'))->toEqual(['one', 'two', 'three']);
    });
  });
  
  describe('with \' one , two , three \'', function() {
    it('produces the expected result', function() {
      expect(Loaders::commaSeparated(' one , two , three '))->toEqual(['one', 'two', 'three']);
    });
  });
  
  describe('with \',,\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::commaSeparated(',,'))->toEqual(['', '', '']);
    });
  });
  
  describe('with \'one two three\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::commaSeparated('one two three'))->toEqual(['one two three']);
    });
  });
  
  describe('with \'one;two;three\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::commaSeparated('one;two;three'))->toEqual(['one;two;three']);
    });
  });
  
  describe('with \'   \'', function() {
    it('produces the expected result', function() {
      expect(Loaders::commaSeparated('   '))->toEqual(['']);
    });
  });
});

File: ruby/spec/comma_separated.spec.rb
require 'enotype'

describe 'comma_separated' do
  context 'with \'one,two,three\'' do
    it 'produces the expected result' do
      expect(Enotype::comma_separated('one,two,three')).to eq(['one', 'two', 'three'])
    end
  end
  
  context 'with \' one,two,three \'' do
    it 'produces the expected result' do
      expect(Enotype::comma_separated(' one,two,three ')).to eq(['one', 'two', 'three'])
    end
  end
  
  context 'with \'one , two , three\'' do
    it 'produces the expected result' do
      expect(Enotype::comma_separated('one , two , three')).to eq(['one', 'two', 'three'])
    end
  end
  
  context 'with \' one , two , three \'' do
    it 'produces the expected result' do
      expect(Enotype::comma_separated(' one , two , three ')).to eq(['one', 'two', 'three'])
    end
  end
  
  context 'with \',,\'' do
    it 'produces the expected result' do
      expect(Enotype::comma_separated(',,')).to eq(['', '', ''])
    end
  end
  
  context 'with \'one two three\'' do
    it 'produces the expected result' do
      expect(Enotype::comma_separated('one two three')).to eq(['one two three'])
    end
  end
  
  context 'with \'one;two;three\'' do
    it 'produces the expected result' do
      expect(Enotype::comma_separated('one;two;three')).to eq(['one;two;three'])
    end
  end
  
  context 'with \'   \'' do
    it 'produces the expected result' do
      expect(Enotype::comma_separated('   ')).to eq([''])
    end
  end
end